diff --git a/src/javascript/mraajs.i b/src/javascript/mraajs.i
index 321cc22..4d9949c 100644
--- a/src/javascript/mraajs.i
+++ b/src/javascript/mraajs.i
@@ -42,18 +42,18 @@ namespace mraa {
 class Spi;
 %typemap(out) uint8_t*
 {
-%#if SWIG_V8_VERSION > 0x040000
+%#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION <= 0x032870)
+  $result = node::Buffer::New((char*) result, arg3)->handle_;
+  free(result);
+%#elif (V8_MAJOR_VERSION-0) < 4
+  $result = node::Buffer::New((char*) result, arg3);
+  free(result);
+%#else
    v8::MaybeLocal<v8::Object> objret = node::Buffer::Copy(v8::Isolate::GetCurrent(), (char*) result, arg3);
    free(result);
    if(!objret.ToLocal(&$result)){
 	SWIG_exception_fail(SWIG_ERROR, "Spi buffer failed");
    }
-%#elif SWIG_V8_VERSION > 0x032870
-  $result = node::Buffer::New((char*) result, arg3);
-  free(result);
-%#else
-  $result = node::Buffer::New((char*) result, arg3)->handle_;
-  free(result);
 %#endif
 }
 }
@@ -81,18 +81,18 @@ class Spi;
        free($1);
        SWIG_exception_fail(SWIG_ERROR, "Uart write failed");
    }
-%#if SWIG_V8_VERSION > 0x040000
-   v8::MaybeLocal<v8::Object> objret = node::Buffer::Copy(v8::Isolate::GetCurrent(), (char*) $1, result);
+%#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION <= 0x032870)
+   $result = node::Buffer::New((char*) $1, result)->handle_;
    free($1);
-   if(!objret.ToLocal(&$result)) {
-      SWIG_exception_fail(SWIG_ERROR, "Uart buffer failed");
-   }
-%#elif SWIG_V8_VERSION > 0x032870
+%#elif (V8_MAJOR_VERSION-0) < 4
    $result = node::Buffer::New((char*) $1, result);
    free($1);
 %#else
-   $result = node::Buffer::New((char*) $1, result)->handle_;
+   v8::MaybeLocal<v8::Object> objret = node::Buffer::Copy(v8::Isolate::GetCurrent(), (char*) $1, result);
    free($1);
+   if(!objret.ToLocal(&$result)) {
+      SWIG_exception_fail(SWIG_ERROR, "Uart buffer failed");
+   }
 %#endif
 
 }
@@ -116,18 +116,18 @@ class Spi;
        free($1);
        SWIG_exception_fail(SWIG_ERROR, "I2c write failed");
    }
-%#if SWIG_V8_VERSION > 0x040000
-   v8::MaybeLocal<v8::Object> objret = node::Buffer::Copy(v8::Isolate::GetCurrent(), (char*) $1, result);
+%#if (V8_MAJOR_VERSION-0) < 4 && (SWIG_V8_VERSION <= 0x032870)
+   $result = node::Buffer::New((char*) $1, result)->handle_;
    free($1);
-   if(!objret.ToLocal(&$result)) {
-      SWIG_exception_fail(SWIG_ERROR, "I2c buffer failed");
-   }
-%#elif SWIG_V8_VERSION > 0x032870
+%#elif (V8_MAJOR_VERSION-0) < 4
    $result = node::Buffer::New((char*) $1, result);
    free($1);
 %#else
-   $result = node::Buffer::New((char*) $1, result)->handle_;
+   v8::MaybeLocal<v8::Object> objret = node::Buffer::Copy(v8::Isolate::GetCurrent(), (char*) $1, result);
    free($1);
+   if(!objret.ToLocal(&$result)) {
+      SWIG_exception_fail(SWIG_ERROR, "I2c buffer failed");
+   }
 %#endif
 }
 
