--- a/fileio.c
+++ b/fileio.c
@@ -23,6 +23,13 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifndef DEFFILEMODE
+#define DEFFILEMODE (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)
+#endif
+#ifndef MAXNAMLEN
+#define MAXNAMLEN 255
+#endif
+
 #include "def.h"
 #include "kbd.h"
 #include "pathnames.h"
--- a/re_search.c
+++ b/re_search.c
@@ -21,6 +21,10 @@
 #include <stdio.h>
 #include <string.h>
 
+#ifndef REG_STARTEND
+#define REG_STARTEND 0
+#endif
+
 #include "def.h"
 #include "macro.h"
 
